### PR TITLE
Integrating poetry into github actions

### PR DESCRIPTION
**Pull request by Kong-Codes**

### CODE REVIEW

File: .github/workflows/gspread-cicd.yml [hunk 1]
```diff
@@ -23,26 +22,33 @@ jobs:
...
-        make format
+        cd project-3
+        poetry run black tests/unit/*.py ./cli.py ./exceptions.py ./main.py
         
     - name: Lint code with pylint
       run: |
-        make lint
-  
+        cd project-3
+        poetry run pylint tests/unit/*.py ./cli.py ./exceptions.py ./main.py
```
Comment (elseagle) [inline .github/workflows/gspread-cicd.yml:45]: Set a failure threshold - your action shows many Pylint warnings, but your action still says passed despite multiple warnings.

File: .github/workflows/gspread-cicd.yml [hunk 2]
```diff
@@ -23,26 +22,33 @@ jobs:
...
+      continue-on-error: true
+
     - name: Test with pytest
       run: |
-        make test
+        cd project-3
+        poetry run pytest tests/unit/test_gspread_utility.py 
         
     - name: Deploy to server
```
Comment (elseagle) [inline .github/workflows/gspread-cicd.yml]: Why do you have this task? What server is being deployed to?

**Review by elseagle — CHANGES_REQUESTED** [review]
Good start @Kong-Codes 

I see commits from you only, why were the mentors not notified about the absence of your teammates? This was meant to be a team work.

File: .github/workflows/gspread-cicd.yml (hunk 2 above)
Comment (Kong-Codes) [reply .github/workflows/gspread-cicd.yml]: I have removed it

**Review by elseagle — APPROVED** [review]

**Pull request merged**